readthedocs/search/faceted_search.py

```diff
@@ -1,15 +1,30 @@
 from elasticsearch_dsl import FacetedSearch, TermsFacet
 
 
-class ProjectSearch(FacetedSearch):
-    fields = ['name^5', 'description']
-    facets = {
-        'language': TermsFacet(field='language')
-    }
+class RTDFacetedSearch(FacetedSearch):
+    """Overwrite the initialization in order too meet our needs"""
+
+    # TODO: Remove the overwrite when the elastic/elasticsearch-dsl-py#916
+    # See more: https://github.com/elastic/elasticsearch-dsl-py/issues/916
 
     def __init__(self, using, index, doc_types, model, **kwargs):
         self.using = using
         self.index = index
         self.doc_types = doc_types
         self._model = model
-        super(ProjectSearch, self).__init__(**kwargs)
+        super(RTDFacetedSearch, self).__init__(**kwargs)
+
+
+class ProjectSearch(RTDFacetedSearch):
+    fields = ['name^5', 'description']
+    facets = {
+        'language': TermsFacet(field='language')
+    }
+
+
+class FileSearch(RTDFacetedSearch):
+    fields = ['title^10', 'headers^5', 'content']
+    facets = {
+        'project': TermsFacet(field='project'),
+        'version': TermsFacet(field='version')
+    }
```

Comment:
Love how simple this is.

Reply:
Yes. me too! the thing is made so simple by using OOP concepts!